### PR TITLE
fix event handling for choosing config from launcher

### LINKED ISSUE #2414: [BUG] Clicking a config icon in the config launcher no longer does anything
### Jdaviz component

Other (explain in Description below)

### Description

Some time semi-recently, the config launcher stopped working. Running the command line `jdaviz` command will start the config launcher as expected, and you can import data and have it filter the configs. But actually clicking a config icon no longer does anything. @camipacifici seems to remember it first not working when she installed #2408.

### How to Reproduce

Install dev version of jdaviz, run `jdaviz` from command line and try to open a config (optionally after importing data).

### Expected behavior

_No response_

### Browser

_No response_

### Jupyter

jupyter --version

Selected Jupyter core packages...
IPython          : 8.14.0
ipykernel        : 6.25.1
ipywidgets       : 8.1.0
jupyter_client   : 7.4.9
jupyter_core     : 5.3.1
jupyter_server   : 2.7.2
jupyterlab       : 3.6.5
nbclient         : 0.7.4
nbconvert        : 7.8.0
nbformat         : 5.9.2
notebook         : 6.5.5
qtconsole        : 5.4.3
traitlets        : 5.9.0

### Software versions

_No response_

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request fixes the event handling on the click event in the launcher.

Note: v3.6 does not have this full version of the launcher, so this should not need to be backported.

Fixes #2414

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/core/launcher.vue [hunk 1]
```diff
@@ -76,7 +76,7 @@
         class="mx-3"
         color="#FFFFFF"
         style="height: 180px; width: 115px"
-        @click="launch_config(config)"
+        @click="launch_config({config: config})"
```
Comment (kecnry) [inline jdaviz/core/launcher.vue:79]: I have no idea how this ever worked... I'm guessing it was changed at one point but anyone testing with an editable install was not getting the latest version 🤷.  Reviewers - you may need a fresh env to force the templates to be updated in your installation.

**Comment by rosteen** [comment]
This works when I run `jdaviz` from the command line and then immediately click a config. However, if I load a file and then click the config it highlights as relevant, nothing happens still (for example loading an x1d file and then clicking Specviz).

**Comment by pllim** [comment]
@rosteen , that's weird. Worked for me when I tested with Imviz (load file and then click with Imviz the only thing enabled). I used Windows 10 + Chrome, but I did not have Notebook 7.

**Comment by rosteen** [comment]
> @rosteen , that's weird. Worked for me when I tested with Imviz (load file and then click with Imviz the only thing enabled). I used Windows 10 + Chrome, but I did not have Notebook 7.

It's nothing to do with Notebook 7, I see the same thing with 6.x. Looks like it's actually Specviz-specific, I see Specviz failing to load with multiple input file (so it's not just a bad file), but I do actually see Imviz working on my machine as well with an i2d file selected.

**Comment by rosteen** [comment]
Maybe there's a bug in the way `Specviz.load_data()` is getting called? I know that originally Specviz used `load_spectra` and `load_data` there is more recent 🤔

**Comment by rosteen** [comment]
> Maybe there's a bug in the way `Specviz.load_data()` is getting called? I know that originally Specviz used `load_spectra` and `load_data` there is more recent 🤔

I think this is it, looks like `Specviz.load_data` doesn't have a **kwargs input handler and probably chokes on the 'verbosity' and 'history_verbosity' keywords input from jdaviz_cli.ipynb.

**Comment by pllim** [comment]
Loading the same file directly in notebook using `specviz.load_data(...)` works, so the bug must be in the launch somewhere.

**Comment by pllim** [comment]
I don't understand why the launcher insists on using `hdul` from

https://github.com/spacetelescope/jdaviz/blob/15772497f2ac103a34b3a92a10a39af478ccfdf4/jdaviz/core/data_formats.py#L270

and not passing in the original input `filename`. That function is `data_formats.py` is bypassing real parser logic, tricking Specviz into thinking it is loading a SpectrumList, and then Specviz crashes.

File: jdaviz/core/launcher.py [hunk 2]
```diff
@@ -82,7 +86,12 @@ def _launch_config_with_data(config, data=None, show=True, **kwargs):
 
     # Load data
     if data not in (None, ''):
-        viz_helper.load_data(data, **kwargs)
+        try:
+            viz_helper.load_data(data, **kwargs)
+        except IORegistryError:
+            if filepath is None:
+                raise
+            viz_helper.load_data(filepath, **kwargs)
```
Comment (rosteen) [inline jdaviz/core/launcher.py:94]: Not the prettiest thing I've ever seen but I don't hate it. And it works, which is the important thing!

**Review by rosteen — APPROVED** [review]
Specviz now loads data successfully like the rest of the configs, approved. Thanks!

File: jdaviz/core/launcher.py (hunk 2 above)
Comment (kecnry) [reply jdaviz/core/launcher.py:94]: agreed... hopefully we can revert this down the road, but that may require some significant refactoring [:cat:](https://jira.stsci.edu/browse/JDAT-3755)

**Pull request merged**